core/include/keep.S

```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2014, STMicroelectronics International N.V.
```

Comment:
What @etienne-lms proposes is exactly to define a assembly macros in a `.h` file ;-) and I don't see why it would be impossible. Do you have any example in mind? Would the pre-processor behave differently when it is working for the assembler rather than the C compiler? I don't think so, except that the command-line flags may be different of course.

In any case, I tried @etienne-lms 's suggestion and it's working fine, so I'll push an update.